todo_app/app.py

```diff
@@ -1,14 +1,52 @@
-from flask import Flask
-
+from flask import abort, Flask, render_template, request, redirect, session, url_for
+from todo_app.data.session_items import add_item, delete_item, get_item, get_items, save_item
 from todo_app.flask_config import Config
 
 app = Flask(__name__)
 app.config.from_object(Config)
 
-
-@app.route('/')
+@app.route('/', methods=['POST', 'GET'])
 def index():
-    return 'Hello World!'
+    items = get_items()
+        
+    if request.method == 'GET':
+        items.sort(key=is_checked)
+        return render_template('index.html', items=items)
+    elif request.method == 'POST':
+        if new_item := request.form['field_name']:
+            add_item(new_item)
+
+        if deleted_item_ids := [key[7:] for key in request.form.keys() if key.startswith('delete_')]:
+            for id in deleted_item_ids:
+                delete_item(id)
+            return redirect('/')
+
+        all_ids = [str(item['id']) for item in items]
+        checked_item_ids = request.form.getlist('item')
+        checked_items = [get_item(id) for id in checked_item_ids]
+        unchecked_items = [get_item(id) for id in all_ids if id not in checked_item_ids]
+
+        for item in checked_items:
+            item['checked'] = True
+            save_item(item)
+
+        for item in unchecked_items:
+            item['checked'] = False
+            save_item(item)
```

Comment:
Is this equivalent to the below?
```suggestion
        checked_item_ids = request.form.getlist('item')
        for id in all_ids:
            item = get_item(id)
            item['checked'] = id in checked_item_ids
            save_item(item)
```